Source/Diagnostics/ReducedDiags/ParticleNumber.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2019-2020 Neil Zaim, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleNumber.H"
+#include "WarpX.H"
+
+// constructor
+ParticleNumber::ParticleNumber (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
```

Comment:
Can this be `const` ?
```suggestion
    const auto nSpecies = mypc.nSpecies();
```

Reply:
Yes I think these can be const, thanks! For cons(t)istency I'll also add the same consts in `ParticleEnergy.cpp` and `ParticleHistogram.cpp`

Reply:
In the end I added much more consts than just these two files. Hope it's not an issue (but I guess if this introduces an error it should be caught at compile time no?)